fsspec/implementations/smb.py

```diff
@@ -128,7 +156,26 @@ def _port(self):
     def _connect(self):
         import time
 
-        for _ in range(self.register_session_retries):
+        if self.register_session_retries <= -1:
```

Comment:
Is there no way to connect, then? Perhaps should have `_connect(self, retries=None)` and allow overriding the original value if the method is to be called later.

Reply:
I assumed that _connect is private method only used for connecting on initialization. Also i tried to not change the actual behavior as much as possible, only target the issue at hand (compatibility issues). 
Since it is possible to connect later using fs methods and kwargs, i just places the option to not connect ( which was there since register_session_retries implementation ).

Reply:
Here, i am just making assumptions about fsspec and set consistencies for its features. Just by looking at SMBFileSystem implementation, i can se that each of its methods call smbclient which tries to find registered session OR create new session using passed kwargs. Following example is a working usage:
```python
from fsspec.implementations.smb import SMBFileSystem
smb = SMBFileSystem(host="mysmbhost", register_session_retries=-1)
...
smb.listdir("/sharefolder", username="user", password="password")
```
I do not know why would anyone wanted to use it like this. Maybe it could be useful where credentials are supplied at later time (i am here really only speculating). Either way i just followed already present option to not connect by setting retries to 0.

Reply:
I didn't realise that smbclient had an auto-connect if it hadn't already been done. Now I'm worried what might happen if a user wanted to connect to multiple SMB services, but let's leave that for now.